src/test/java/com/spotify/autoscaler/FakeBTCluster.java

```diff
@@ -63,14 +72,53 @@ public FakeBTCluster(final Supplier<Instant> timeSource, final BigtableCluster c
     return metrics;
   }
 
+  public Instant getFirstMetricsInstant() {
+    final Map.Entry<Instant, ClusterMetricsData> firstMetrics =
+        metrics
+            .entrySet()
+            .stream()
+            .sorted(Comparator.comparing(Map.Entry::getKey))
+            .filter(e -> e.getValue().nodeCount() > 0)
```

Comment:
I just had to check in the sample files what this is about. I see nodeCount is sometimes 0.0 and I guess it's StackDriver metrics weirdness (UDP?). Seems to happen with cpu and diskUtil too. You should put a comment here why you are filtering on `nodeCount > 0`.